### DISCUSSION TITLE
Formatting section not generated

### DISCUSSION THREAD

**Discussion by MaximilianGierlachowski** [detekt/detekt#4896] [Q&A] [answered]
Hey, sorry that I am asking questions here, but I just can't get it to work so I wanted to confirm it isn't a bug and ask how it should be done. Basically I am using detekt version 1.20.0. I am applying the plugin in my app module with the plugin syntax like so:
`id("io.gitlab.arturbosch.detekt")`
I don't provide a version because detekt is added to the classpath already by the buildSrc build.gradle.kts with a specific version like so (shouldn't make a difference for the issue):
`implementation("io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.20.0")`
then in the app module I am trying to apply the formatting plugin in the dependency block like so:
`detektPlugins("io.gitlab.arturbosch.detekt:detekt-formatting:1.20.0")`

When I try to generate the config there is no formatting section inside the config, shouldn't there be one?

Edit:

For context when I don't apply the formatting plugin and use an old config where the formatting block was part of the config I get and error that it isn't recognized, with the plugin is applied like above I don't get that error. Meaning the plugin gets applied successfully but does not take part in generating the config file.

Is this expected behaviour and one should add all the formatting rules themself? Because this seems a little bit tedious.

**Comment by cortinico** [comment, marked as answer]
That's a known bug in 1.20.0. Has been addressed here https://github.com/detekt/detekt/pull/4844
Will be fixed in 1.21 👍

**Reply by MaximilianGierlachowski** [reply]
Ohh nice thanks a lot, sorry I missed that one when I was searching 😅
